base/simd.jl

```diff
@@ -80,4 +79,26 @@ end
     return nothing
 end
 
+import Base: +, -, *, /, muladd, promote_rule, widen
+import Core.Intrinsics: add_float, sub_float, mul_float, div_float, muladd_float, neg_float
+
+## floating point promotions ##
+promote_rule(::Type{Vec{N, Float32}}, ::Type{Vec{N, Float16}}) where N = Vec{N, Float32}
```

Comment:
It is not obvious to me that these should be defined. When you are doing low level SIMD stuff you probably don't want to accidentally promote things and in case where you really want to work with different types, an explicit convert might be better for clarity?